float64 lat
float64 long
float64 X
float64 Y
